### PR TITLE
podio, edm4hep: Add latest versions

### PR DESCRIPTION
**Pull request by tmadlener**
Latest tags for both packages

@vvolkl

### CODE REVIEW

**Review by vvolkl — DISMISSED** [review]

**Review by vvolkl — DISMISSED** [review]

File: var/spack/repos/builtin/packages/podio/package.py [hunk 1]
```diff
@@ -115,6 +116,10 @@ def cmake_args(self):
     def setup_run_environment(self, env):
         env.prepend_path("PYTHONPATH", self.prefix.python)
         env.prepend_path("LD_LIBRARY_PATH", self.spec["podio"].libs.directories[0])
+        if "+sio" in self.spec and self.version >= Version("0.16"):
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/podio/package.py:119]: FYI. We prefer using `self.spec.satisfies()` for non-variant checks, such as version (ranges).  (See https://spack.readthedocs.io/en/latest/packaging_guide.html#testing-spec-constraints.)

**Review by tldahlgren — APPROVED** [review]
Confirmed both new sha256.

**Comment by tldahlgren** [comment]
@vvolkl Is this PR ready to be merged with the latest change?

**Review by vvolkl — APPROVED** [review]

**Comment by vvolkl** [comment]
> @vvolkl Is this PR ready to be merged with the latest change?

Yes it is. Thanks for the pointer to `self.spec.satisfies()`!

**Pull request merged**